pkg/server/endpoints/registration/handler.go

```diff
@@ -536,6 +536,46 @@ func (h *Handler) FetchBundle(
 	}, nil
 }
 
+//Evict removes (de-attest) node from attested nodes store
+func (h *Handler) EvictAgent(ctx context.Context, evictRequest *registration.EvictAgentRequest) (*registration.EvictAgentResponse, error) {
+	spiffeID := evictRequest.GetSpiffeID()
+	err := h.deleteAttestationEntry(ctx, spiffeID)
+	if err != nil {
+		h.Log.Warnf("Fail to de-attested agent with ID: %q", spiffeID)
+		return &registration.EvictAgentResponse{
+			DeleteSucceed: false,
+		}, err
+	}
+
+	h.Log.Infof("Successfully de-attested agent with ID: %q", spiffeID)
+	return &registration.EvictAgentResponse{
+		DeleteSucceed: true,
+	}, nil
+}
+
+//List returns the list of attested nodes
+func (h *Handler) ListAgents(ctx context.Context, empty *common.Empty) (*registration.ListAgentsResponse, error) {
+	dataStore := h.Catalog.DataStores()[0]
+	listAttestedNodesRequest := &datastore.ListAttestedNodesRequest{}
+	listAttestedNodesResponse, err := dataStore.ListAttestedNodes(ctx, listAttestedNodesRequest)
+	if err != nil {
+		return nil, err
+	}
+	return &registration.ListAgentsResponse{Nodes: listAttestedNodesResponse.Nodes}, nil
+}
+
+func (h *Handler) deleteAttestationEntry(ctx context.Context, baseSPIFFEID string) error {
```

Comment:
nit: it would be nice if this matched the datastore naming convention e.g. `deleteAttestedNode`

Reply:
Another note here, I've been trying to avoid the term `baseSPIFFEID` because I don't think it carries a lot of meaning and can cause confusion... it was a term we coined very early on in the design and development process, but as time goes on, I've found I prefer `agentID` or something along those lines